### ISSUE TITLE
Returning Task in an entry point method of a dnx console app

### ISSUE THREAD

**Issue by kichalla** [aspnet/dnx#2109]
I was using some async apis in my dnx console app and so thought of returning a Task, but returning  a task seems to cause any exception in my main method to not show up on the console and the process just exits without any information...I have [read](http://stackoverflow.com/a/9212343/1184056) that in general we should not be returning Task in an entry point method, but was curious if this rule even applies to a dnx console app?

My main method returning Task:
```c#
public class Program
{
    public async Task Main(string[] args)
    {
        var configurationBuilder = new ConfigurationBuilder();
        configurationBuilder.AddJsonFile("NonExistingFile.json")
                    .AddEnvironmentVariables();
        var configuration = configurationBuilder.Build();
```

Exception while debugging:
```
System.InvalidOperationException occurred
  HResult=-2146233079
  Message=Unable to resolve path 'NonExistingFile.json'; construct this IConfigurationBuilder with a non-null BasePath.
  Source=Microsoft.Framework.Configuration
  StackTrace:
       at Microsoft.Framework.Configuration.Helper.ConfigurationHelper.ResolveConfigurationFilePath(IConfigurationBuilder configuration, String path)
       at Microsoft.Framework.Configuration.JsonConfigurationExtension.AddJsonFile(IConfigurationBuilder configuration, String path, Boolean optional)
       at Microsoft.Framework.Configuration.JsonConfigurationExtension.AddJsonFile(IConfigurationBuilder configuration, String path)
       at SqlServerCacheSample.Program.<Main>d__0.MoveNext() in C:\github\caching\samples\SqlServerCacheSample\Program.cs:line 28
  InnerException:
```

**Comment by Joe4evr** [comment]
Dupe of dotnet/roslyn#1695.

**Comment by davidfowl** [comment]
@kichalla The no async main method doesn't apply to DNX, we always wait at the entry point if Main is async. If error aren't showing up then we might just have a bug.

**Comment by moozzyk** [comment]
I think it should be shown here: https://github.com/aspnet/dnx/blob/32358ec9b8970070ff77cbc5beef28c13e3ab09b/src/dnx.host/RuntimeBootstrapper.cs#L31

**Comment by davidfowl** [comment]
Right

**Comment by moozzyk** [comment]
I was doing some work in this area I looked at this. It appears that when debugging with VS the `DNX_CONSOLE_HOST` environment variable is set to `0` which results in not printing the exception details (and possibly an AV on CoreClr which is a separate issue). We only set `DNX_CONSOLE_HOST` to `1` and only if it is not set. @BillHiebert - does VS set `DNX_CONSOLE_HOST` to 0 when debugging with F5? Why?
